sympy/simplify/simplify.py

```diff
@@ -515,6 +515,10 @@ def simplify(expr, ratio=1.7, measure=count_ops, rational=False, inverse=False):
     kwargs = dict(ratio=ratio, measure=measure,
         rational=rational, inverse=inverse)
 
+    # no routine for Expr needs to check for is_zero
+    if isinstance(expr, Expr) and expr.is_zero and expr*0 is S.Zero:
```

Comment:
Why `expr*0 is S.Zero` is there when `expr.is_zero` is already there?

Reply:
`Matrix.zeros(2).is_zero`

Reply:
but `Matrix.zeros(2) is not S.Zero`...I am just looking for scalar zeros.